contracts/liquidity_hub/bonding-manager/src/helpers.rs

```diff
@@ -0,0 +1,260 @@
+    }
+
+    Ok(())
+}
+
+/// if user has unclaimed rewards, fail with an exception prompting them to claim
+pub fn validate_claimed(deps: &DepsMut, _info: &MessageInfo) -> Result<(), ContractError> {
+    // Do a smart query for Claimable
+    let claimable_rewards: ClaimableEpochsResponse = get_claimable_epochs(deps.as_ref()).unwrap();
+
+    // If epochs is greater than none
+    if !claimable_rewards.epochs.is_empty() {
+        return Err(ContractError::UnclaimedRewards {});
+    }
+
+    Ok(())
+}
+
+/// Validates that the current time is not more than a day after the epoch start time. Helps preventing
+/// global_index timestamp issues when querying the weight.
+pub fn validate_bonding_for_current_epoch(deps: &DepsMut, env: &Env) -> Result<(), ContractError> {
+    let epoch_response: EpochResponse = get_current_epoch(deps.as_ref()).unwrap();
+
+    let current_epoch = epoch_response.epoch;
+    let current_time = env.block.time.seconds();
+    pub const DAY_IN_SECONDS: u64 = 86_400u64;
+
+    // if the current time is more than a day after the epoch start time, then it means the latest
+    // epoch has not been created and thus, prevent users from bonding/unbonding to avoid global_index
+    // timestamp issues when querying the weight.
+    if current_epoch.id != Uint64::zero()
+        && current_time - current_epoch.start_time.seconds() > DAY_IN_SECONDS
+    {
+        return Err(ContractError::NewEpochNotCreatedYet {});
+    }
+
+    Ok(())
+}
+
+/// Calculates the epoch id for any given timestamp based on the genesis epoch configuration.
+pub fn calculate_epoch(
+    genesis_epoch_config: EpochConfig,
+    timestamp: Timestamp,
+) -> StdResult<Uint64> {
+    let epoch_duration: Uint64 = genesis_epoch_config.duration;
+
+    // if this is true, it means the epoch is before the genesis epoch. In that case return Epoch 0.
+    if Uint64::new(timestamp.nanos()) < genesis_epoch_config.genesis_epoch {
+        return Ok(Uint64::zero());
+    }
+
+    let elapsed_time =
+        Uint64::new(timestamp.nanos()).checked_sub(genesis_epoch_config.genesis_epoch)?;
+    let epoch = elapsed_time
+        .checked_div(epoch_duration)?
+        .checked_add(Uint64::one())?;
+
+    Ok(epoch)
+}
+
+// Used in FillRewards to search the funds for LP tokens and withdraw them
+// If we do get some LP tokens to withdraw they could be swapped to whale in the reply
+pub fn handle_lp_tokens(
+    info: &MessageInfo,
+    config: &white_whale_std::bonding_manager::Config,
+    messages: &mut Vec<CosmosMsg>,
+) -> Result<(), ContractError> {
+    let lp_tokens: Vec<&Coin> = info
+        .funds
+        .iter()
+        .filter(|coin| coin.denom.contains(".pair.") | coin.denom.contains(LP_SYMBOL))
+        .collect();
+    for lp_token in lp_tokens {
+        // LP tokens have the format "{pair_label}.pair.{identifier}.{LP_SYMBOL}", get the identifier and not the LP SYMBOL
+        let pair_identifier = lp_token.denom.split(".pair.").collect::<Vec<&str>>()[1];
```

Comment:
I am not sure this work correctly, wouldn't this split it as `{pair_label}` and `{identifier}.{LP_SYMBOL}`? it would need to be split by the `.` and then take the element in index `2` i believe